### ISSUE TITLE
🐛 Bug Report: Aggregate ownership of large groups fail due to p-limit hanging after hitting max concurrent limit in plugin-orgs

### ISSUE THREAD

**Issue by CiscoRob** [backstage/backstage#24530]
### 📜 Description

Recently updated to 1.25.2 and have had some issues most recently with request limits prior to that, which seem to have been patched in this version or prior minor revisions.

The problem we have is in our "root" top level team it has about 10 child groups, and each of those groups could have 1:* groups, and each may have 1:* services. 

The root group doesn't own things directly, so it requires the aggregated relationship view to see the details, but when enabled the URL filter becomes pretty sizeable. Watching the network tab about 12 `by-refs/` calls go out but it never completes loading the data.

From the user perspective they click to view the Group (aka team) and on the details page the data loads but the ownership card just sits there with the linear progress bar looking like it's doing something. Behind the scenes the requests stopped, no console errors, no errors reported in the page, and it won't return back the data.

For us that's a big problem because this happens to be the senior leadership team that _wants_ to see all the aggregated counts of its child teams and instead they're getting a progress bar with no data ever returning.

### 👍 Expected behavior

Ideally the relationship "ownership" card, when set to aggregated view, would always manage to return back the expected data, but this seems to be impacted by rate limits and the depth / size of the organization structure. If we're modeling ownership across hundreds of teams or thousands if counting GitHub team integrations this aggregation needs to be reliable.

### 👎 Actual Behavior with Screenshots

<img width="553" alt="Screenshot 2024-04-26 at 9 59 52 AM" src="https://github.com/backstage/backstage/assets/133238823/761a2520-353a-4ed1-baff-e36916acdb87">

<img width="719" alt="Screenshot 2024-04-26 at 9 51 20 AM" src="https://github.com/backstage/backstage/assets/133238823/966cff51-9eeb-4c57-9097-d1d6f600e6b3">

### 👟 Reproduction steps

Create a sizeable team structure (around 300) and then ensure that the nested hierarchy includes a singular root group that has aggregated view enabled.

With around 1000 services assigned across 300 teams the filter on the root group becomes pretty large and seems to hit the rate limit and just _stops_ trying to request data to build the aggregated view.

Oddly when the https://<domain>/api/catalog/entities/by-refs/ are called it reaches 12 then ... stops. There are a number of larger groups with around 8-10 child teams and always when it gets to the network tab i see exactly 12 `by-refs/` following the `entities?filter...` request before it completely stops.

### 📃 Provide the context for the Bug.

In larger groups within Backstage looking at the details for a parent group to many child teams who all own services results in that parent team's details page unable to ever generate the aggregated view of ownership.

### 🖥️ Your Environment

OS:   Darwin 23.4.0 - darwin/arm64
node: v18.18.0
yarn: 1.22.19
cli:  0.26.2 (installed)
backstage:  1.25.2

Dependencies:
  @backstage/app-defaults                                          1.5.3
  @backstage/backend-app-api                                       0.6.2
  @backstage/backend-common                                        0.21.6
  @backstage/backend-defaults                                      0.2.16
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.9
  @backstage/backend-plugin-api                                    0.6.16
  @backstage/backend-tasks                                         0.5.21
  @backstage/backend-test-utils                                    0.3.6
  @backstage/catalog-client                                        1.6.3
  @backstage/catalog-model                                         1.4.5
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.4
  @backstage/cli                                                   0.24.0, 0.26.2
  @backstage/config-loader                                         1.7.0
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.12.3
  @backstage/core-compat-api                                       0.2.3
  @backstage/core-components                                       0.13.10, 0.14.3
  @backstage/core-plugin-api                                       1.9.1
  @backstage/dev-utils                                             1.0.30
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.6
  @backstage/frontend-app-api                                      0.6.3
  @backstage/frontend-plugin-api                                   0.6.3
  @backstage/integration-aws-node                                  0.1.12
  @backstage/integration-react                                     1.1.25
  @backstage/integration                                           1.9.1
  @backstage/plugin-api-docs                                       0.11.3
  @backstage/plugin-app-backend                                    0.3.64
  @backstage/plugin-app-node                                       0.1.16
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.8
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.8
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.11
  @backstage/plugin-auth-backend-module-github-provider            0.1.13
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.13
  @backstage/plugin-auth-backend-module-google-provider            0.1.13
  @backstage/plugin-auth-backend-module-guest-provider             0.1.2
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.11
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.13
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.9
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.7
  @backstage/plugin-auth-backend-module-okta-provider              0.0.9
  @backstage/plugin-auth-backend                                   0.22.3
  @backstage/plugin-auth-node                                      0.4.11
  @backstage/plugin-auth-react                                     0.0.3
  @backstage/plugin-catalog-backend-module-github-org              0.1.11
  @backstage/plugin-catalog-backend-module-github                  0.5.7
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.14
  @backstage/plugin-catalog-backend                                1.21.0
  @backstage/plugin-catalog-common                                 1.0.22
  @backstage/plugin-catalog-graph                                  0.4.3
  @backstage/plugin-catalog-node                                   1.11.0
  @backstage/plugin-catalog-react                                  1.11.2
  @backstage/plugin-catalog                                        1.18.2
  @backstage/plugin-events-node                                    0.3.2
  @backstage/plugin-org                                            0.6.23
  @backstage/plugin-permission-backend-module-allow-all-policy     0.1.13
  @backstage/plugin-permission-backend                             0.5.40
  @backstage/plugin-permission-common                              0.7.13
  @backstage/plugin-permission-node                                0.7.27
  @backstage/plugin-permission-react                               0.4.21
  @backstage/plugin-proxy-backend                                  0.4.14
  @backstage/plugin-scaffolder-backend-module-azure                0.1.8
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.6
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.6
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.6
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.8
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.6
  @backstage/plugin-scaffolder-backend-module-github               0.2.6
  @backstage/plugin-scaffolder-backend-module-gitlab               0.3.2
  @backstage/plugin-scaffolder-backend                             1.22.3
  @backstage/plugin-scaffolder-common                              1.5.1
  @backstage/plugin-scaffolder-node                                0.4.2
  @backstage/plugin-scaffolder-react                               1.8.3
  @backstage/plugin-scaffolder                                     1.19.2
  @backstage/plugin-search-backend-module-catalog                  0.1.21
  @backstage/plugin-search-backend-module-techdocs                 0.1.21
  @backstage/plugin-search-backend-node                            1.2.20
  @backstage/plugin-search-backend                                 1.5.6
  @backstage/plugin-search-common                                  1.2.11
  @backstage/plugin-search-react                                   1.7.9
  @backstage/plugin-search                                         1.4.9
  @backstage/plugin-techdocs-backend                               1.10.3
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.8
  @backstage/plugin-techdocs-node                                  1.12.2
  @backstage/plugin-techdocs-react                                 1.2.2
  @backstage/plugin-techdocs                                       1.10.3
  @backstage/plugin-user-settings                                  0.8.4
  @backstage/release-manifests                                     0.0.11
  @backstage/test-utils                                            1.5.3
  @backstage/theme                                                 0.4.4, 0.5.2
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by awanlin** [comment]
Hi @CiscoRob, I think this is a lot like this one? https://github.com/backstage/backstage/issues/22139

**Comment by CiscoRob** [comment]
I'm wondering if it's related to p-lint. I see the package set to version 3.1.x and latest for the package is 5.x. Some updates related to async context handling have been made through version 4 and 5. Seems to get hung up after it gets past its maximum concurrent level of 10.

<img width="685" alt="Screenshot 2024-04-26 at 11 33 49 AM" src="https://github.com/backstage/backstage/assets/133238823/14216ba2-abeb-44b1-bf5e-852cd88a4c27">

We're using bottleneck in some parts of our implementation for doing complex lookups for data exports and I've seen similar behavior from other schedulers.

**Comment by CiscoRob** [comment]
@awanlin I pulled Backstage source, checked out v1.26.0-next.1, and imported the `plugins-org` into my local Backstage instance. 

I confirmed / verified the "hanging" behavior was present in my local portal with the package directly referenced. I then removed the `limiter()` from the plugin-orgs and everything worked.

<img width="1246" alt="Screenshot 2024-04-26 at 1 47 26 PM" src="https://github.com/backstage/backstage/assets/133238823/c691d613-12e7-4958-81ae-f5b9a5de7dfd">

The `limiter` is definitely the issue and seems to just ... hang after any call past the 10 maximum concurrent count. Leaves the async action as if it's still "loading" but it never responds.
